### PR TITLE
configure: check for realpath function now MSYS2 has it

### LINKED ISSUE #2261: [Windows] MSYS2 causes `realpath` to be redeclared in `tm_source_file.c`
I'm using MSYS2 as recommended by: https://wiki.geany.org/howtos/win32/msys2.
Building git master.
```
$ gcc --version
gcc (GCC) 7.4.0
```
`make V=1` gives:
```
depbase=`echo tm_source_file.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ../../libtool --silent  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I. -I../../ctags/main -DGEANY_PRIVATE -DG_LOG_DOMAIN=\"Tagmanager\"  -mms-bitfields -pthread -mms-bitfields -I/mingw32/include/gtk-2.0 -I/mingw32/lib/gtk-2.0/include -I/mingw32/include/pango-1.0 -I/mingw32/include/fribidi -I/mingw32/include -I/mingw32/include/cairo -I/mingw32/include/atk-1.0 -I/mingw32/include/cairo -I/mingw32/include/pixman-1 -I/mingw32/include -I/mingw32/include/freetype2 -I/mingw32/include -I/mingw32/include/harfbuzz -I/mingw32/include -I/mingw32/include/libpng16 -I/mingw32/include/gdk-pixbuf-2.0 -I/mingw32/include/libpng16 -I/mingw32/include -I/mingw32/lib/libffi-3.2.1/include -I/mingw32/include -I/mingw32/include/glib-2.0 -I/mingw32/lib/glib-2.0/include -I/mingw32/include -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_32 -fvisibility=hidden -DGEANY_EXPORT_SYMBOL="__attribute__((visibility(\"default\")))" -DGEANY_API_SYMBOL=GEANY_EXPORT_SYMBOL -g -DGEANY_DEBUG -g -O0 -MT tm_source_file.lo -MD -MP -MF $depbase.Tpo -c -o tm_source_file.lo tm_source_file.c &&\
mv -f $depbase.Tpo $depbase.Plo
tm_source_file.c:91:14: error: static declaration of ‘realpath’ follows non-static declaration
 static char *realpath (const char *pathname, char *resolved_path)
              ^~~~~~~~
In file included from tm_source_file.c:20:0:
/usr/include/stdlib.h:151:8: note: previous declaration of ‘realpath’ was here
 char * realpath (const char *__restrict path, char *__restrict resolved_path);
        ^~~~~~~~
make[3]: *** [Makefile:505: tm_source_file.lo] Error 1
make[3]: Leaving directory '/c/git/geany/src/tagmanager'
```

### PR DESCRIPTION
**Pull request by ntrel**
Fixes #2261.

### CODE REVIEW

File: src/tagmanager/tm_source_file.c [hunk 1]
```diff
@@ -84,7 +84,7 @@ static int get_path_max(const char *path)
 }
 
 
-#ifdef G_OS_WIN32
+#if defined(G_OS_WIN32) && !defined(HAVE_REALPATH)
```
Comment (codebrainz) [inline src/tagmanager/tm_source_file.c]: You could probably drop the `G_OS_WIN32` part altogether, but this is probably safer for now.

Comment (ntrel) [reply src/tagmanager/tm_source_file.c]: Well, the implementation only compiles on Windows.

**Comment by b4n** [comment]
Is this still relevant?  Looking at the conversation in #2261 it's kind of unclear.
However, this looks good anyhow.

**Comment by kugel-** [comment]
Should probably merge this for 1.36?

**Comment by elextr** [comment]
Two approvals is enough for me.

**Pull request merged**